teuthology/run.py

```diff
@@ -168,10 +168,15 @@ def validate_tasks(config):
 
 
 def get_initial_tasks(lock, config, machine_type):
-    init_tasks = [
-        {'internal.check_packages': None},
-        {'internal.buildpackages_prep': None},
-    ]
+    init_tasks = []
+    overrides = config.get('overrides')
+    having_repos = ('install' in config and 'repos' in config.get('install')) \
```

Comment:
I don’t think I understand what’s happening here. I presume you want to bypass these internal tasks if not in the upstream lab, but the conditions don’t make a lot of sense to me.

Reply:
Yes, the idea we don't want to build ceph packages ourself if we provided test repos.